llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -900,6 +900,41 @@ namespace {
                          ISD::NodeType ExtType);
   };
 
+/// Generic remainder optimization : Folds a remainder operation (A % B) by reusing the computed quotient (A / B).
+static SDValue PerformREMCombineGeneric(SDNode *N, DAGCombiner &DC,
```

Comment:
Passing in DAGCombiner is strange, pass in the DAG and TLI, or just move to be a DAGCombiner member?